ui/lesson/ScriptingChallenge/index.tsx

```diff
@@ -66,32 +72,34 @@ export default function ScriptingChallenge({
   const t = useTranslations(lang)
   const { saveProgress, saveProgressLocal } = useProgressContext()
   const { account } = useAuthContext()
+  const { currentLanguage, setCurrentLanguage } = useDataContext()
   const [code, setCode] = useState(
-    config.languages[config.defaultLanguage].defaultCode?.toString()
+    config.languages[getLanguageString(currentLanguage)].defaultCode?.toString()
   )
   const [constraints, setConstraints] = useState(
-    config.languages[config.defaultLanguage].constraints
+    config.languages[getLanguageString(currentLanguage)].constraints
   )
 
   const [hiddenRange, setHiddenRange] = useState(
-    config.languages[config.defaultLanguage].hiddenRange
+    config.languages[getLanguageString(currentLanguage)].hiddenRange
   )
-  const [language, setLanguage] = useState(config.defaultLanguage)
+  const [language, setLanguage] = useState(getLanguageString(currentLanguage))
+  console.log(language)
   const [challengeSuccess, setChallengeSuccess] = useState(false)
   const [hydrated, setHydrated] = useState(false)
   const [errors, setErrors] = useState<string[]>([])
   const debouncedCode = useDebounce(code, 500)
 
   useEffect(() => {
-    const savedCode = localStorage.getItem(lessonKey)
+    const savedCode = localStorage.getItem(`${lessonKey}-${language}`)
```

Comment:
@benalleng this is the fix for the saved code issue I mentioned.  Adding the language to the local storage key makes it unique for each challenges language selection.